<launch>
  <arg name="uav_name" default="$(optenv UAV_NAME uav1)"/>
  <!-- Node Settings -->
  <arg name="debug"             default="false"/>
  <arg name="output_frame"      default="local_origin"/>
  <arg name="use_velocity"      default="false"/>
  <arg name="output_framerate"  default="10"/>

  <arg name="indoor"            default="false"/>
  <!-- <arg name="use_two_cameras"   default="true"/> -->
  <!-- <arg name="use_three_cameras"   default="false"/> -->
  <arg name="anonymous_measurements"   default="false"/>

  <group ns="$(arg uav_name)">
  
    <!-- <node pkg="uvdar_core" type="uvdar_filter_node" name="filterUvdar" output="screen" launch-prefix="debug_roslaunch"> -->
      <node pkg="uvdar_core" type="uvdar_filter_node" name="uvdar_filter" output="screen">
        <param name="debug"             value="$(arg debug)" />
        <param name="uav_name"          value="$(arg uav_name)" />
        <param name="output_frame"      value="$(arg output_frame)"/>
        <param name="output_framerate"  value="$(arg output_framerate)" />

        <param name="indoor"            value="$(arg indoor)" />
        <param name="use_velocity"      value="$(arg use_velocity)" />
        <param name="anonymous_measurements"      value="$(arg anonymous_measurements)" />

        <rosparam param="measured_poses_topics"> ["measured_poses"] </rosparam>
        <remap from="~measured_poses" to="/$(arg uav_name)/uvdar/measuredPoses"/>

        <remap from="~filtered_poses" to="/$(arg uav_name)/uvdar/filteredPoses"/>
        <remap from="~filtered_poses/tentative" to="/$(arg uav_name)/uvdar/filteredPoses/tentative"/>

    </node>
  </group>

</launch>
